src/PdfViewer/index.tsx

```diff
@@ -0,0 +1,93 @@
+import 'react-pdf/dist/esm/Page/AnnotationLayer.css';
+import 'react-pdf/dist/esm/Page/TextLayer.css';
+
+import { FileEntry } from "@tauri-apps/api/fs";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { Document, Page } from "react-pdf";
+
+import { readFile } from "../filesystem";
+import { PdfViewerAPI } from '../types/PdfViewerAPI';
+
+export interface PdfViewerProps {
+  file: FileEntry;
+  pdfViewerRef: React.MutableRefObject<PdfViewerAPI | null>;
+}
+
+export const PdfViewer = ({ file, pdfViewerRef }: PdfViewerProps) => {
```

Comment:
I think we should use a function declaration for react components instead of this syntax.